pennylane/ftqc/pauli_tracker.py

```diff
@@ -117,24 +128,74 @@ def pauli_prod(ops: List[Operator]) -> Operator:
             from pennylane import I, X, Y, Z
             >>> pauli_prod([I(0),X(0),Y(0),Z(0)])
-            I(0)
+            (0, 0)
 
         A Pauli operator is returned for a list of Pauli operator up to a global phase.
     """
 
     if len(ops) == 0:
         raise ValueError("Please ensure that a valid list of operators are passed to the method.")
-    op0 = ops.pop()
-    res_x, res_z = pauli_to_xz(op0)
-    op0_wire = op0.wires
+    res_x, res_z = pauli_to_xz(ops.pop())
 
     while len(ops) > 0:
-        op = ops.pop()
-        wire = op.wires
-        if wire != op0_wire:
-            raise ValueError("All operators should target at the same wire.")
-        x, z = pauli_to_xz(op)
+        x, z = pauli_to_xz(ops.pop())
         res_x ^= x
         res_z ^= z
 
-    return xz_to_pauli(res_x, res_z)(wires=op0_wire)
+    return (res_x, res_z)
+
+
+def apply_clifford_op(
+    clifford_op: Operator, xz: List[Tuple[np.uint8, np.uint8]]
+) -> List[Tuple[np.uint8, np.uint8]]:
+    """Commuting a list of Pauli ops to a list of new Pauli ops with a given Clifford op.
+
+    Args:
+        clifford_op (Operator): A Clifford operator class. Supported operators are: :class:`qml.S`, :class:`qml.H`, :class:`qml.CNOT`.
+        xz (list(tuple)): A list of xz tuples which map to Pauli operators
+
+    Return:
+        A list of new xz tuples that the clifford_op commute the xz to.
+
+    **Example:**
+        The following example shows how the `pauli_prod` works.
+
+        .. code-block:: python3
+
+            from pennylane.ftqc.pauli_tracker import apply_clifford_op
+            from pennylane import I, CNOT
+            >>> apply_clifford_op(CNOT(wires=[0,1]), [(0, 0), (0, 0)])
+            [(0, 0), (0, 0)]
+
+        A list of xz representation of Pauli operators is returned.
+    """
+    if type(clifford_op) not in _CLIFFORD_TABLEAU:
+        raise NotImplementedError("Only qml.H, qml.S and qml.CNOT are supported.")
+
+    if len(xz) != clifford_op.num_wires:
+        raise ValueError(
+            "Please ensure that the length of xz matches the number of wires of the clifford_op."
+        )
+
+    if not all(len(element) == 2 for element in xz):
+        raise ValueError(
+            "Please ensure there are 2 elements instead of in each tuple in the xz list."
+        )
+
+    xz_flatten = tuple(itertools.chain.from_iterable(xz))
+
+    if not all(element in [0, 1] for element in xz_flatten):
+        raise ValueError("Please ensure xz are either 0 or 1.")
+
+    if all(element == (0, 0) for element in xz):
+        return xz
+
+    # A Clifford gate conjugate non-Identify Pauli ops to a new Pauli ops
+    new_xz = []
+    nonzero_indices = [idx for idx, element in enumerate(xz_flatten) if element == 1]
+
+    # Get Paulis prod for each target wire
+    for table_row in _CLIFFORD_TABLEAU[type(clifford_op)]:
+        ps = [table_row[idx] for idx in nonzero_indices]
+        new_xz.append(pauli_prod(ps))
+    return new_xz
```

Comment:
Same query here if we are retuning uint8 or just regular integers

Reply:
Updated to int. Thanks